api/src/main/resources/plugins/oozie.py

```diff
@@ -48,6 +48,23 @@ def validate_component(self, component):
     def get_component_type(self):
         return 'oozie'
 
+    def assert_application_properties(self, override_properties, default_properties):
```

Comment:
This function needs a refactor, too much duplicated code. I would first merge the properties then have a single loop that checks the spark version.

Reply:
Also looking closer why iterate a map checking for a key, can't you do a single map lookup for the key?